### ISSUE TITLE
How to make an observer which react to events related to programmatically generated inputs with variable names?

### ISSUE THREAD

**Issue by PMarmalyuk** [rstudio/shiny#1015]
Hi all!
First of all I want to thank the developers of Shiny: great job!

Now I want to describe my issue and ask for help of the community.

I am trying to create an example of a ShinyApp that allows to automatically generate UI inputs for some scalars provided in a named list. For example, I feed my App with this list: 

    method <- list('Method name' = "My Great Method", 
                          'Post process' = TRUE,
                          'Iterations' = 20).

The App should scan this list and create an input for each element of the list. Ids of the inputs should be set to be equal to the tranformed names of the elements in the list (e.g. lowercase and without whitespace). This part is solved programmatically and I have a reactive list which contains ids of all inputs created.

For the example above the App's UI should include three inputs: a text input with id "uiFrame_methodname", a checkboxInput with id "uiFrame_postprocess" and a numerical input with id "uiFrame_iterations". So in this case I have access to, say, a vector `vars` on the server side. It keeps ids of created input elements ("uiFrame_methodname", "uiFrame_postprocess" and "uiFrame_iterations").

User, of course, is able to change values of these inputs. And the App should keep track of it: when a user changes value of any of the inputs, I need to trigger the event handler. Usually I use observe() to do something like this, but it works only when I explicitly specify an input id which should be observed by Shiny:
 
    observe({
    if (input$mySliderInput1 > 200)
    {print("Hello")}
    })

If I try to implement such observer dependency implicitly, Shiny fails to keep track of changes in corresponding inputs (observer does not react), e.g.:

    observe({
    if (input[[vars[1]]] == "My Great Method")
    {print("Hello")}
    })

So, in short, I need to make Shiny react to events related to programmatically generated inputs whose names are kept inside some variable.

I hope I described everything clearly and recieve an answer soon! 
Thank you in advance!

**Comment by daattali** [comment]
You might want to post this on the [shiny Google group forum](https://groups.google.com/forum/#!forum/shiny-discuss) since that's where people ask for advice, whereas GitHub Issues are more meant to be a place to report bugs

**Comment by jcheng5** [comment]
I'm guessing the `vars` vector itself isn't a reactive variable. You'll have to make it so by calling `makeReactiveBinding(vars)`.

If that's not it and you'd like to continue the conversation, please start a thread on the Shiny Google group as Dean said. Thanks!

**Comment by PMarmalyuk** [comment]
Thank you for advice and sorry for posting my question in inappropriate place.
